src/components/organisms/NewProfileModal/components/links/ProfileModalEditLinks/ProfileModalEditLinks.tsx

```diff
@@ -15,51 +15,43 @@ import "./ProfileModalEditLinks.scss";
 
 export interface ProfileModalEditLinksProps extends ContainerClassName {
   initialLinks: ProfileLink[];
-  links: Partial<ArrayField<ProfileLink>>[];
-  setLinkTitle: (index: number, title: string) => void;
-  register: FormFieldProps["register"];
-  errors?: NestDataObject<ProfileLink, FieldError>[];
+  links: ProfileLink[];
+  register: UseFormRegister<UserProfileModalFormData>;
+  errors?: FieldErrors<ProfileLink>[];
   onDeleteLink: (index: number) => void;
   onAddLink: () => void;
 }
 
 export const ProfileModalEditLinks: React.FC<ProfileModalEditLinksProps> = ({
   initialLinks,
   links,
-  setLinkTitle,
   register,
   errors,
   onDeleteLink,
   onAddLink,
   containerClassName,
 }) => {
-  const setTitles = useMemo(
-    () => links.map((_, i) => (title: string) => setLinkTitle(i, title)),
-    [links, setLinkTitle]
-  );
-
   const renderedLinks = useMemo(
     () =>
       register &&
-      links.map(({ id, ...link }, i) => {
+      links.map(({ ...link }, i) => {
         const otherUrls = links.filter((l) => l !== links[i]).map((l) => l.url);
 
         return (
           <ProfileModalEditLink
             containerClassName="ProfileModalEditLinks__link-group"
-            key={id}
+            key={i}
```

Comment:
renamed to `index`, but not going to use any `link` value since they're dynamic and React not working well when trying to edit a field with a dynamic key

Reply:
> they're dynamic and React not working well when trying to edit a field with a dynamic key
Well, this is an interesting issue, would be interested to learn more about the problem after we merge this PR